src/components/web3-connect/index.tsx

```diff
@@ -69,6 +69,22 @@ export class Container extends React.Component<Properties, State> {
     this.props.setConnectionStatus(ConnectionStatus.Disconnected);
   }
 
+  deactivateConnector() {
+    if (localStorage) {
+      localStorage.removeItem('previousConnector');
+    }
+    this.props.updateConnector(Connectors.Infura);
+    this.props.setAddress('');
```

Comment:
i think we can simplify this. can we just pull the `setAddress` call out, and base it on account change?
eg. i think setAddress can always just happen like this:
```
    if (web3.account !== prevAccount) {
      this.props.setAddress(web3.account);
    }
```
that way we're not missing any calls by trying to call it in every place we think it might be needed.